src/es/components/modules/tabs/Tabs.js

```diff
@@ -0,0 +1,96 @@
+// @ts-check
+'use strict'
+
+import { Shadow } from '../../prototypes/Shadow.js'
+
+/**
+ * Creates Tabs
+ * adapted from https://gist.github.com/ebidel/2d2bb0cdec3f2a16cf519dbaa791ce1b
+ * @export
+ * @attribute {selected} preselected tab index (default the first tab is selected)
+ * @type {CustomElementConstructor}
+ */
+export default class Tabs extends Shadow() {
+  constructor () {
+    super({ mode: 'open' })
+
+    this.onClick = (e) => {
+      const target = e.target.slot === 'tabBtn' ? e.target : e.target.closest('[slot=tabBtn]')
+
+      if (target.slot === 'tabBtn') {
+        this.selectedElement = this.tabBtns.indexOf(target)
+        target.focus()
+      }
+    }
+  }
+
+  connectedCallback () {
+    if (this.shouldComponentRenderHTML()) this.renderHTML()
+
+    this.tabBtns.forEach((btn, index) => {
+      btn.addEventListener('click', this.onClick)
+
+      // set initial state
+      if (this.selectedElement === index) this.selectedElement = index
```

Comment:
does this change anything?

Reply:
it triggers the setter for selectedElement